### PR TITLE
Yk IR: Implement stringification of common constant integers.

### PR DESCRIPTION
**Pull request by vext01**
(Writing the tests for this one was 90% of the work!)

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/aot_ir.rs [hunk 1]
```diff
@@ -547,4 +591,72 @@ func bar;
...
+        use num_traits::cast::NumCast;
+        use std::mem;
+
+        // Check (in an endian neutral manner) that a `num-bits`-sized integer of value `num`, when
+        // converted to a constant IR integer, then stringified, results in the string `expect`.
+        //
+        // When `num` has a bit size greater than `num_bits` the most significant bits of `num` are
+        // treated as undefined: they can be any value as IR stringification will ignore them.
+        fn check<T: NumCast + Sized>(num_bits: u32, num: T, expect: &str) {
+            dbg!(mem::size_of::<T>() * 8, usize::try_from(num_bits).unwrap());
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/aot_ir.rs:606]: Debugging code.

File: ykrt/src/compile/jitc_yk/aot_ir.rs [hunk 2]
```diff
@@ -547,4 +591,72 @@ func bar;
...
+            assert!(mem::size_of::<T>() * 8 >= usize::try_from(num_bits).unwrap());
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/aot_ir.rs:607]: `debug_assert` ?

File: ykrt/src/compile/jitc_yk/aot_ir.rs [hunk 3]
```diff
@@ -547,4 +591,72 @@ func bar;
...
+                2 => bytes
+                    .write_u16::<NativeEndian>(<u16 as NumCast>::from(num).unwrap())
+                    .unwrap(),
+                4 => bytes
+                    .write_u32::<NativeEndian>(<u32 as NumCast>::from(num).unwrap())
+                    .unwrap(),
+                8 => bytes
+                    .write_u64::<NativeEndian>(<u64 as NumCast>::from(num).unwrap())
+                    .unwrap(),
+                _ => panic!("{}", mem::size_of::<T>()),
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/aot_ir.rs:624]: `todo` or `unreachable` ?

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 2 above)
Comment (vext01) [reply ykrt/src/compile/jitc_yk/aot_ir.rs:607]: it's in a test, so i figured we may as well always check.

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 3 above)
Comment (vext01) [reply ykrt/src/compile/jitc_yk/aot_ir.rs:624]: I never know which to use.

This is something that we will need to implement one day, but is impossible under the current assumptions of the system. You tell me :P

Comment (ltratt) [reply ykrt/src/compile/jitc_yk/aot_ir.rs:624]: `todo` means "I will fix this later". `unreachable` means "if you've got here I've got something wrong". `panic` means "if you've got here, the caller has done something wrong".

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 2 above)
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/aot_ir.rs:607]: Ah!

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 3 above)
Comment (vext01) [reply ykrt/src/compile/jitc_yk/aot_ir.rs:624]: So it's a mix of `todo` and `unreachable`.

`unreachable` for now, `todo` in the future.

Inclined to go with `todo` so I don't have to (forget to) flip a `unreachable` to a `todo` later.

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 1 above)
Comment (vext01) [reply ykrt/src/compile/jitc_yk/aot_ir.rs]: Oops. Fixed in 27ece7c3abb3

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 3 above)
Comment (vext01) [reply ykrt/src/compile/jitc_yk/aot_ir.rs]: Fixed in 27ece7c3abb3

**Comment by vext01** [comment]
Ready for re-review.

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**